include/dsn/dist/distributed_lock_service.h

```diff
@@ -69,7 +69,7 @@ namespace dsn
             /*
              * initialization routine
              */
-            virtual error_code initialize() = 0;
+            virtual error_code initialize(const char* work_dir, const char* lock_root) = 0;
```

Comment:
You are right about passing any parameters in it is impossible mission for the callers. To pass any possible explicit setting is also impossible as we don't know what kind of settings the underlying provider want. As a result, to me the best solution so far is leave it no parameters, and the providers should use configurations to get the wanted parameters. That way, we leave the provider-specific things in config instead of code, which makes things more clean. @qinzuoyan